src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -65,7 +65,7 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
         // be preferable not to have another version of that lying around. Until we really hit a scenario where floating point formatting needs the perf, we'll
         // make do with this.
         //
-        private static bool TryFormatFloatingPoint<T>(T value, Span<byte> destination, out int bytesWritten, StandardFormat format) where T : IFormattable
+        private static bool TryFormatFloatingPoint<T>(T value, Span<byte> destination, out int bytesWritten, StandardFormat format) where T : ISpanFormattable
```

Comment:
Now that the `Utf8Formatter` lives in `S.P.Corelib`, the comment on this method isn't really applicable anymore.

We should just be able to call the existing `DoubleToNumber()` directly and then call the UTF8 `NumberToString` (now that the code is shared, `NumberBuffer` always has a `Span<byte>`, so we shouldn't need any conversion, etc).

Reply:
This would get rid of the allocations and would likely make the code more efficient overall.

Reply:
Now I'm a bit confused, this ``DoubleToNumber()`` in https://github.com/dotnet/coreclr/blob/356a08f7e90113b31005c02abb14662bc40bed70/src/System.Private.CoreLib/shared/System/Number.Formatting.cs#L2338?
Isn't that bit too deep? All the special format specifier handling is done higher in the call stack.
I'd understand https://github.com/dotnet/coreclr/blob/356a08f7e90113b31005c02abb14662bc40bed70/src/System.Private.CoreLib/shared/System/Number.Formatting.cs#L371-L379

But that would still work on 'char' internally, or do you mean rewriting Number.Formatting to work on utf8?

Reply:
NumberBuffer is an abstract type that is independent of the source/destination encoding (UTF8 or UTF16). 

For formatting, the `XToNumber` methods take the input and produces a sequence of ASCII digits that represent that number and also tracks some additional info (like the sign, and "scale" of the original value). The resulting `NumberBuffer` is then passed into `NumberToString`, which will return a `Span<char>` or `Span<byte>` containing the properly formatted string; inserting the sign, decimal point, and additional information (such as the exponent) as necessary. I don't think we have a Utf8 version of `NumberToString`, which is the additional step that would be required.

For parsing, we basically do the same, but in reverse.

Reply:
Thank you for your explanation. Looking at the code, besides the special cases, like NaN, most of the code directly appends the ASCII symbols to the ValueStringBuilder. Why not having an ASCIIValueStringBuilder working on byte-arrays with a flag which increments the index twice for chars? That might be easier than NumberToString for Utf8.